tests/validation/tests/v3_api/common.py

```diff
@@ -643,18 +645,31 @@ def wait_for_pod_to_running(client, pod, timeout=DEFAULT_TIMEOUT):
     return p
 
 
-def get_schedulable_nodes(cluster, client=None, os_type=TEST_OS):
+def get_schedulable_nodes(cluster, client=None, os_type=TEST_OS, activeNodes=False):
     if not client:
         client = get_user_client()
     nodes = client.list_node(clusterId=cluster.id).data
     schedulable_nodes = []
-    for node in nodes:
-        if node.worker and (not node.unschedulable):
-            for key, val in node.labels.items():
-                # Either one of the labels should be present on the node
-                if key == 'kubernetes.io/os' or key == 'beta.kubernetes.io/os':
-                    if val == os_type:
-                        schedulable_nodes.append(node)
+    if activeNodes:
+        for node in nodes:
+            if node.worker and (not node.unschedulable) and node.state == "active":
```

Comment:
instead of having if and else separately with same logic, we can add `activeNodes` in this condition. 
so it becomes, 
```
// by default, assumption is all nodes are active unless the bool variable says to actually check 
nodeActive = True
if activeNodes:
   nodeActive = node.state == "active" 
if nodeActive and node.worker and ... 
```